contracts/IdentityRegistryBridger.sol

```diff
@@ -0,0 +1,328 @@
+    }
+
+    /// @dev Registers 8004 agent Id corresponding to service Id.
+    /// @param serviceId Service Id.
+    /// @return agentId Corresponding 8004 agent Id.
+    function _register(uint256 serviceId) internal returns (uint256 agentId) {
+        // Get token URI
+        string memory tokenUri = IERC721(serviceRegistry).tokenURI(serviceId);
+
+        // Get actual service multisig
+        (,address multisig,,,,,) = IServiceRegistry(serviceRegistry).mapServices(serviceId);
+
+        // Check for zero address, although this must never happen
+        if (multisig == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // TODO Work on this block to finalize
+        // Assemble OLAS specific metadata entry
+        IIdentityRegistry.MetadataEntry[] memory metadataEntries = new IIdentityRegistry.MetadataEntry[](1);
+        metadataEntries[0] = IIdentityRegistry.MetadataEntry({
+            key: "Ecosystem",
+            value: abi.encode("OLAS V1")
+        });
+
+        // Create new agent Id
+        agentId = IIdentityRegistry(identityRegistry).register(tokenUri, metadataEntries);
+
+        // Link service Id and agent Id
+        mapServiceIdAgentIds[serviceId] = agentId;
+
+        // Approve service multisig such that it becomes agentId operator
+        IERC721(identityRegistry).approve(multisig, agentId);
+
+        emit AgentRegistered(serviceId, agentId, multisig, tokenUri);
+    }
+
+    /// @dev Updates 8004 agent Id corresponding to service Id.
+    /// @param serviceId Service Id.
+    /// @param agentId Corresponding 8004 agent Id.
+    function _update(uint256 serviceId, uint256 agentId) internal {
+        // Get token URI
+        string memory serviceTokenUri = IERC721(serviceRegistry).tokenURI(serviceId);
+
+        // Get actual service multisig
+        (,address serviceMultisig,,,,,) = IServiceRegistry(serviceRegistry).mapServices(serviceId);
+
+        // Check for zero address, although this must never happen
+        if (serviceMultisig == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // Get 8004 agent Id tokenUri
+        string memory agentTokenUri = IERC721(identityRegistry).tokenURI(agentId);
+        // Check if tokenUri has changed
+        if (keccak256(bytes(serviceTokenUri)) != keccak256(bytes(agentTokenUri))) {
+            // Set updated tokenUri
+            IIdentityRegistry(identityRegistry).setAgentUri(agentId, serviceTokenUri);
+        }
+
+        // Check for multisig change
+        address agentServiceMultisig = IERC721(identityRegistry).getApproved(agentId);
+        // Check if multisig address has changed
+        if (serviceMultisig != agentServiceMultisig) {
+            // Approve updated service multisig such that it becomes agentId operator
+            IERC721(identityRegistry).approve(serviceMultisig, agentId);
+        }
+
+        // TODO Shall we NOT emit anything if nothing has been changed?
```

Comment:
Yeah not sure, at the end the data are the same, and we do not need to update essentially nothing. Though, for symmetry, I would keep it